tests/integration/API/JSON/v1/FormsTest.php

```diff
@@ -186,40 +192,41 @@ public function testItCanExtractListsFromNestedListSelectionBlock() {
   }
 
   public function testItCanRestoreAForm() {
-    $this->form1->trash();
+    $this->form1->setDeletedAt(new \DateTime());
+    $this->formsRepository->flush();
 
-    $trashedForm = Form::findOne($this->form1->id);
+    $trashedForm = Form::findOne($this->form1->getId());
     assert($trashedForm instanceof Form);
     expect($trashedForm->deletedAt)->notNull();
 
-    $response = $this->endpoint->restore(['id' => $this->form1->id]);
+    $response = $this->endpoint->restore(['id' => $this->form1->getId()]);
     expect($response->status)->equals(APIResponse::STATUS_OK);
     expect($response->data)->equals(
-      $this->reloadForm((int)$this->form1->id)->asArray()
+      $this->reloadForm((int)$this->form1->getId())->asArray()
     );
```

Comment:
`$this->reloadForm` internally uses the old model class. It is something that we want to get rid of. Looking at the whole expect, it makes not much sense to me. We compare that the response, that is created by a [specialized response builder class](https://github.com/mailpoet/mailpoet/blob/master/lib/API/JSON/ResponseBuilders/FormsResponseBuilder.php) is equal to an old model instance converted to an array. The response builder has its own integration test so I guess we could either drop the response format assert from this test or if we want to check that API returns the response built by the response builder we could add one test case for this into this test. Does that make sense to you? What do you think?

Reply:
> $this->reloadForm internally uses the old model class. It is something that we want to get rid of.

Some of the \MailPoet\API\JSON\v1\Forms methods that are being tested in this class still use the old form model. I believe it will be easier to change \MailPoet\Test\API\JSON\v1\FormsTest::reloadForm() once we update all the methods inside the class being tested so outside of the scope of this PR. Sounds good?

> We compare that the response, that is created by a specialized response builder class is equal to an old model instance converted to an array. The response builder has its own integration test so I guess we could either drop the response format assert from this test or if we want to check that API returns the response built by the response builder we could add one test case for this into this test. Does that make sense to you? What do you think?

I tend to agree with your assessment of this test, but I'm inclined to suggest that we address this in a separate PR as it seems outside of the scope of this one and it will probably be easier to address it once all methods of the class being tested are migrated to the new model. Once that happens, we might be able to remove reloadForm(). What do you think?